chia/wallet/util/wallet_sync_utils.py

```diff
@@ -41,7 +41,8 @@ async def fetch_last_tx_from_peer(height: uint32, peer: WSChiaConnection) -> Opt
         if response is not None and isinstance(response, RespondBlockHeader):
             if response.header_block.is_transaction_block:
                 return response.header_block
-        else:
+        elif request_height < height:
```

Comment:
Do you think we could get stuck in an infinite loop here? (Well, "infinite" for as long as the other peer is behind our current height)

Reply:
No because we do decrement the request height for every iteration of the loop, so as soon as we get either:
1. a tx block
2. a negative response

we break

Reply:
Ah, yes, we fall through out of the if block in that case, thanks.